src/ripple/app/misc/ValidatorList.h

```diff
@@ -0,0 +1,99 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2015 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_APP_MISC_VALIDATORLIST_H_INCLUDED
+#define RIPPLE_APP_MISC_VALIDATORLIST_H_INCLUDED
+
+#include <ripple/basics/Log.h>
+#include <ripple/basics/UnorderedContainers.h>
+#include <ripple/core/Config.h>
+#include <ripple/protocol/PublicKey.h>
+#include <boost/optional.hpp>
+#include <functional>
+#include <memory>
+#include <mutex>
+
+namespace ripple {
+
+class ValidatorList
+{
+private:
+    /** The non-ephemeral public keys from the configuration file. */
+    hash_map<PublicKey, std::string> permanent_;
+
+    /** The ephemeral public keys from manifests. */
+    hash_map<PublicKey, std::string> ephemeral_;
+
+    std::mutex mutable mutex_;
+    beast::Journal mutable j_;
+
+public:
+    ValidatorList (beast::Journal j);
+
+    /** Determines whether a node is in the UNL
+        @return empty optional if the node isn't a member,
+                otherwise, the comment associated with the
+                node (which may be empty).
```

Comment:
Instead of "empty optional", I'd suggest either "uninitialized optional" (`optional` has an `is_initialized` member) or "boost::none". Then instead of "(which may be empty)", maybe "(which may be an empty string)"